### DISCUSSION TITLE
Please let me know the FLOPS of the LDPCBPdecoder of Sionna

### DISCUSSION THREAD

**Discussion by JWooram** [NVlabs/sionna#267] [Q&A]
Hello. I am a student who is working on a project in Korea under the theme of improving GNN-based decoders using sionna. I'm inquiring because I can't find the FLOPS of LDPCBPdecoder provided by Sionna during the project. I'd like to know the FLOPS for that algorithm. If you don't know, would it be effective to write a pytorch and measure it?

**Comment by SebastianCa** [comment]
Hi,
Great to hear that you are working with Sionna.
The exact number of FLOPs depends on many parameters such as the parity-check matrix, the number of iterations and the exact realization of the node updates. Sionna does not directly provide tools to measure the FLOPS. However, as Sionna is based on TensorFlow, you can utilize TensorFlow’s built-in tools to profile your code. For more information, please refer to [TF.Profiler](https://www.tensorflow.org/api_docs/python/tf/compat/v1/profiler/Profiler).

Please note, it’s important to be aware that GNN-based decoders, like those in Sionna, typically have a higher computational complexity compared to traditional "classical" BP decoders. This is due to the more complex node update functions in the GNN architecture.
